plugins/woocommerce/src/Blocks/BlockTypes/ProductGallery.php

```diff
@@ -77,7 +77,7 @@ function( $carry, $item ) {
 
 		$gallery_dialog = strtr(
 			'
-		<div class="wc-block-product-gallery-dialog__overlay" hidden data-wc-bind--hidden="!context.isDialogOpen" data-wc-watch="callbacks.keyboardAccess">
+		<div class="wc-block-product-gallery-dialog__overlay" hidden data-wc-bind--hidden="!context.isDialogOpen" data-wc-watch="callbacks.keyboardAccess" data-wc-watch--dialog-focus-trap="callbacks.dialogFocusTrap">
```

Comment:
This is mostly because in the API reference it mentions some use cases for the `wc-watch` directive:
> As a reference, some use cases for this directive may be:
> 
> Logging.
> Changing the title of the page.
> Setting the focus on an element with .focus().
> Changing the state or context when certain conditions are met.

One of the reasons is that we want to focus on an element in this case. The other reason is because we want more granular control over when to attach the event, we can have some situations where we don't want to keep track of the keyboard events, like when the pop-up is not available.